example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,194 @@
+    private Map<String, String> recordRules;
+
+    /**
+     * Constructor with 0 arguments for an example implementation
+     * of the {@link PolicyCacheWarmerFactory} interface
+     */
+    public ExamplePolicyCacheWarmerFactory() {
+    }
+
+    /**
+     * Constructor with 4 arguments for an example implementation
+     * of the {@link PolicyCacheWarmerFactory} interface
+     *
+     * @param type          a {@link String} of the {@link Policy} type.
+     * @param owner         a {@link String} value of the owner of the policy
+     * @param resourceRules a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     * @param recordRules   a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     */
+    public ExamplePolicyCacheWarmerFactory(final String type, final String owner, final Map<String, String> resourceRules, final Map<String, String> recordRules) {
+        this.type = type;
+        this.owner = owner;
+        this.resourceRules = resourceRules;
+        this.recordRules = recordRules;
+    }
+
+    @Generated
+    public String getType() {
+        return type;
+    }
+
+    @Generated
+    public void setType(final String type) {
+        requireNonNull(type);
+        this.type = type;
+    }
+
+    @Generated
+    public String getOwner() {
+        return owner;
+    }
+
+    @Generated
+    public void setOwner(final String owner) {
+        requireNonNull(owner);
+        this.owner = owner;
+    }
+
+    @Generated
+    public Map<String, String> getResourceRules() {
+        return resourceRules;
+    }
+
+    @Generated
+    public void setResourceRules(final Map<String, String> resourceRules) {
+        requireNonNull(resourceRules);
+        this.resourceRules = resourceRules;
+    }
+
+    @Generated
+    public Map<String, String> getRecordRules() {
+        return recordRules;
+    }
+
+    @Generated
+    public void setRecordRules(final Map<String, String> recordRules) {
+        requireNonNull(recordRules);
+        this.recordRules = recordRules;
+    }
+
+    @Override
+    public Policy<Employee> policyWarm() {
+        Policy<Employee> policy = new Policy<>();
+        if ("Alice".equals(getOwner())) {
+            policy.owner(ExampleUsers.getAlice());
+        }
+        if ("Bob".equals(getOwner())) {
+            policy.owner(ExampleUsers.getBob());
+        }
+        if ("Eve".equals(getOwner())) {
+            policy.owner(ExampleUsers.getEve());
+        }
+        for (String key : resourceRules.keySet()) {
+            createRule(policy, key, resourceRules.get(key));
+        }
+        for (String key : recordRules.keySet()) {
+            createRule(policy, key, recordRules.get(key));
+        }
+
+        return policy;
+    }
+
+    private void createRule(final Policy<Employee> policy, final String message, final String rule) {
+        switch (rule) {
+            case "BankDetailsRule":
```

Comment:
any chance of using reflection, such that the rule String is the full class name and therefore we don't need to modify this class to add new rules in future to the example. This would also need you to bring in a flag for whether it is a record or resource rule.

Reply:
I've added some reflection and changed the rule to be the full class name in the yaml

Reply:
The full class name is now used in the yaml. This then uses Class.forName to get the constructor and create a new instance of the rule. A flag is also passed to help define the rule that is returned